### PR TITLE
pandas 2.0 design drafts

### PR DESCRIPTION
**Pull request by wesm**
This is a revised version of the docs at pydata/pandas#13944

### CODE REVIEW

**Comment by wesm** [comment]
I'd like to merge this and take up more detailed discussions in separate issues, that work for everybody?

File: source/internal-architecture.rst [hunk 1]
```diff
@@ -0,0 +1,827 @@
...
+
+* **Signed Integers**
+
+  - Logical: ``Int8/16/32/64``
+  - Physical: ``numpy.int8/16/32/64`` array plus nullness bitmap
+
+* **Unsigned Integers**
+
+  - Logical: ``Int8/16/32/64``
+  - Physical: ``numpy.int8/16/32/64`` array plus nullness bitmap
```
Comment (takluyver) [inline source/internal-architecture.rst]: Does this bit need some `u`s?

File: source/removals.rst [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+===========================
+ Other miscellaneous ideas
+===========================
+
+Dropping Python 2 support
+=========================
+
+With Python 2.7 reaching its supported end-of-life in 2020, like some other
+Python projects (e.g. IPython / Jupyter) we should seriously contemplate making
+pandas 2.0 only support Python 3.5 and higher. In addition to lowering the
```
Comment (takluyver) [inline source/removals.rst]: :+1: even if you decide not to do it for 2.0, I'd love to see pandas come up with a plan for going Python 3 only, and sign https://python3statement.github.io/

**Comment by jorisvandenbossche** [comment]
> I'd like to merge this and take up more detailed discussions in separate issues, that work for everybody?

+1

File: source/internal-architecture.rst (hunk 1 above)
Comment (wesm) [inline source/internal-architecture.rst]: indeed, thank you

File: source/removals.rst (hunk 2 above)
Comment (wesm) [inline source/removals.rst]: given that I'd like to see pandas 2.0 revamp its internal string handling, doing this work for both Python 2 and 3 seems like not a good use of time. would be great to remove all the compatibility code, too (and potentially start making use of some python 3 only features, like asyncio)

**Comment by jorisvandenbossche** [comment]
Note: the built version of the sphinx source is now hosted at https://pydata.github.io/pandas-design/docs/ (and will be automatically updated at each commit to master using doctr (https://drdoctr.github.io/doctr/), https://github.com/pydata/pandas-design/commit/896f9d1ed0125297ac6bf90f170f55f04605eea3)

**Pull request closed without merging**